allennlp/service/predictors/semantic_role_labeler.py

```diff
@@ -61,16 +69,20 @@ def predict_json(self, inputs: JsonDict) -> JsonDict:
         spacy_doc = self.nlp(sentence)
         words = [token.text for token in spacy_doc]
         results = {"words": words, "verbs": []}  # type: JsonDict
-        text = TextField(words, token_indexers=self.token_indexers)
         for i, word in enumerate(spacy_doc):
             if word.pos_ == "VERB":
+                verb = word.text
                 verb_labels = [0 for _ in words]
                 verb_labels[i] = 1
-                verb_indicator = SequenceLabelField(verb_labels, text)
-                output = self.model.tag(text, verb_indicator)
+                instance = self._dataset_reader.text_to_instance(words, verb_labels)
+                output = self._model.forward_on_instance(instance)
+                transition_matrix = self._model.get_viterbi_pairwise_potentials()
+
+                predictions = torch.from_numpy(output['class_probabilities'])
+                max_likelihood_sequence, _ = viterbi_decode(predictions, transition_matrix)
+                tags = [self._model.vocab.get_token_from_index(x, namespace="labels")
+                        for x in max_likelihood_sequence]
```

Comment:
The fact that below, you have kept `return sanitize(results)` seems like an argument for removing `_predictions_to_json`.